client/python/gradio_client/client.py

```diff
@@ -992,7 +992,8 @@ def _inner(*data):
             if self.client.upload_files:
                 data = self.serialize(*data)
             predictions = _predict(*data)
-            predictions = self.process_predictions(*predictions)
+            if self.client.download_files:
```

Comment:
I don't understand this part.

Shouldn't `remove_skipped_components` and `reduce_singleton_output` happen even if the client is not downloading the files to the local compute?

Reply:
The problem is that `remove_skipped_components()` actually should not happen if you are doing `gr.load()`. The `remove_skipped_components()` will not return values for `gr.State()` to users who are using the Client manually, but we do need values for `gr.State()` to be returned when doing `gr.load()`

Reply:
The `reduce_singleton_output()` should, in principle, always be done, but the way we've implemented it here (skipping over certain components) is not compatible with `gr.load()`. I'll see if I can refactor it to be clearer

Reply:
Yea you're right that `remove_skipped_components` should not be happening with `gr.load`.